redesign/base_generator.py

```diff
@@ -9,9 +9,13 @@ class BaseGenerator(metaclass=abc.ABCMeta):
     def __init__(self, profile, options):
         """Initialize generator object."""
         self.profile = profile.report()
-        self.options = options
+
+        if isinstance(options, dict):
+            self.options = options
+        else:
+            self.options = {}
 
     @abc.abstractmethod
-    def synthesize(self):
```

Comment:
any reason for the name change? also, I think we should add a req var for `sample_size`

Reply:
open to diff names for `samples` as well

Reply:
that was actually my recommendation .... but looking at it now `run()` isn't explicitly clear what it does... so I think keeping as `synthesize` is a good idea